utils/zerovec/src/varzerovec/serde.rs

```diff
@@ -163,18 +163,21 @@ mod test {
     use crate::{VarZeroSlice, VarZeroVec};
 
     #[derive(serde::Serialize, serde::Deserialize)]
+    #[expect(dead_code)]
     struct DeriveTest_VarZeroVec<'data> {
         #[serde(borrow)]
         _data: VarZeroVec<'data, str>,
     }
 
     #[derive(serde::Serialize, serde::Deserialize)]
+    #[expect(dead_code)]
```

Comment:
Nit: comment why

Reply:
because we test the drive macro but never construct these

Reply:
I meant add a comment in the code

Reply:
so I put a comment on the `DeriveTest` structs saying "only tests derive macro"?

Reply:
Yes

Reply:
`#[expect(dead_code)] // tests derive macro`

Reply:
I think tests like this should just have a blanket allow

Reply:
You can use `#[expect(dead_code, reason = "foobar")]` btw.

We should move all our code over to `reason` at some point.